lib/charms/loki_k8s/v1/loki_push_api.py

```diff
@@ -2313,6 +2313,92 @@ def _containers(self) -> Dict[str, Container]:
         return {cont: self._charm.unit.get_container(cont) for cont in self._logs_scheme.keys()}
 
 
+class PebbleLogForwarder:
+    """Forward the StdOut output to one or multiple Loki endpoints."""
+
+    def __init__(
+        self,
+        charm: CharmBase,
+        *,
+        relation_name: str = DEFAULT_RELATION_NAME,
+        loki_endpoints: Optional[List[str]] = None,
+    ):
+        self._charm = charm
+        self._relation_name = relation_name
+        self.topology = JujuTopology.from_charm(charm)
+
+        if loki_endpoints is None:
+            self.loki_endpoints = self.fetch_endpoints()
+        else:
+            self.loki_endpoints = loki_endpoints
+
+    def _build_log_target(self, endpoint, index, enable=False):
+        """Build a log target for the log forwarding Pebble layer."""
+        dest_name = f"loki{index}"
+        services_value = ["all"] if enable else ["-all"]
+
+        return {
+            dest_name: {
+                "override": "merge",
+                "type": "loki",
+                "location": endpoint,
+                "services": services_value,
+                "labels": {
+                    "product": "Juju",
+                    "charm_name": self.topology._charm_name,
+                    "model": self.topology._model,
+                    "model_uuid": self.topology._model_uuid,
+                    "application": self.topology._application,
+                    "unit": self.topology._unit,
+                },
```

Comment:
Minor change to make the labels consistent with our usual charm metrics:

```suggestion
                "labels": {
                    "product": "Juju",
                    "charm": self.topology._charm_name,
                    "juju_model": self.topology._model,
                    "juju_model_uuid": self.topology._model_uuid,
                    "juju_application": self.topology._application,
                    "juju_unit": self.topology._unit,
                },
```